### ISSUE TITLE
add visualizations 

### ISSUE THREAD

**Issue by bibhabasumohapatra** [bibhabasumohapatra/Vision_Transformer_from_scratch#1]
torch.nn.CosineSimilarity(dim=(-1), eps=1e-08)(q,k).shape   
Explore what happens when you find cosine similarity visualizations?
